source/github-events/on-profile-dropdown-load.ts

```diff
@@ -1,7 +1,8 @@
 import oneMutation from 'one-mutation';
+import elementReady from 'element-ready';
 
 export default async function onProfileDropdownLoad(): Promise<Element> {
-	const dropdown = document.querySelector('.Header-item:last-child .dropdown-menu')!;
-	await oneMutation(dropdown, {childList: true});
-	return dropdown;
+	const dropdown = await elementReady('.Header-item:last-child .dropdown-menu', {waitForChildren: false});
+	await oneMutation(dropdown!, {childList: true});
```

Comment:
I think at least one of these must be changed:

- `waitForChildren: false` means that the promise might resolve before the `include-fragment` inside has been parsed, so `oneMutation` might just catch that when it appears in the DOM, but that's not the element we want.
- `oneMutation` probably needs a `filter` function to ensure that the mutation actually contains the element we expect.
- `oneMutation` might never be called if the code is executed too late and the mutation already happened.

What if we take a shortcut and use `selector-observer` instead of both? Maybe something like:


```suggestion
	const dropdown = await new Promise(resolve => {
		const observer = observe('the final dropdown content', {
			add(element) {
				observer.disconnect();
				resolve(element);
			}
		})
	})
```

Reply:
Actually it might be an overkill. How about
```suggestion
	const dropdown = await elementReady('.Header-item:last-child .dropdown-menu');
	if (!exists()) {
		await oneMutation(dropdown!, {childList: true, filter: exists});		
	}
```

and then:

```js
function exists() {
	return select.exists('the final dropdown content')
}
```